### PR TITLE
Wildcard enhancement, toggle read in order, prompt queue. 通配符增强，切换顺序读取。

### PR DESCRIPTION
**Pull request by xhoxye**
It can generate multiple images of different prompt words at one time according to the sequence of prompt word queue
代码来自 刁璐璐。
通配符增强，通过勾选切换通配符读取方法，默认不勾选为随机读取一行，勾选后为按顺序读取，并使用相同的种子。
QQ截图20240105192916
QQ截图20240105192943
QQ截图20240105192953

### CODE REVIEW

**Comment by mashb1t** [comment]
just for reference: achieves the same goal as https://github.com/lllyasviel/Fooocus/pull/1503 without having to explicitly prompt all the keywords

**Review by mashb1t — CHANGES_REQUESTED** [review]

**Comment by xhoxye** [comment]
My English is not very good, I use translation software to dialogue, my code comments are to make myself understand, otherwise I may have to re-understand when I look back later

File: modules/async_worker.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import threading
+# xhoxye
```
Comment (mashb1t) [inline modules/async_worker.py:2]:
```suggestion
```

**Comment by mashb1t** [comment]
@xhoxye no worries, that's why there are also community members who can support you there :)
For my part only one line is remaining to be changed as personal usernames as comments are not used throughout the code, but other than that the feature looks quite good and seems ready for code review by an authorised reviewer. I assume that this kind of MR falls into the category of "non-trivial changes" and may be reviewed by mid/end of January.

**Comment by LordMilutin** [comment]
Any tips on how to use this? 
I see the checkbox for wildcards, I just don't know what formats to use in the prompt description box.
Can you give an example of 3 simple prompts?

**Review by mashb1t — APPROVED** [review]

**Comment by xhoxye** [comment]
> the prompt description box

the prompt description box: __mylikeprompt__
Fooocus\wildcards\mylikeprompt.txt
mylikeprompt.txt
QQ截图20240107063916
QQ截图20240107063828
QQ截图20240107064039

**Comment by SerhiiSfr** [comment]
Where is checkbox "upload mask"?????????!!!!!!

**Comment by SerhiiSfr** [comment]
неу! Where is checkbox "upload mask"?????????!!!!!

**Comment by pacmanincarnate** [comment]
In the spirit of keeping Fooocus more user-friendly, I'd recommend replacing the wildcard file reference with a text field in the UI that users can type or paste a list into for a wildcard. It's always kind of a pain to go back and forth between the app and the folder to remember what you named the wildcard file and likely edit one to suit your current needs.

**Comment by xhoxye** [comment]
> In the spirit of keeping Fooocus more user-friendly, I'd recommend replacing the wildcard file reference with a text field in the UI that users can type or paste a list into for a wildcard. It's always kind of a pain to go back and forth between the app and the folder to remember what you named the wildcard file and likely edit one to suit your current needs.

Are you looking for this?
QQ截图20240130164633

QQ截图20240130164645

**Comment by bin862324915** [comment]
Previously, I used version 2.1.865 and it worked fine. However, after updating to version 2.2.1, I found that this feature was no longer available. Therefore, I added your code separately to the corresponding files, and when I started it, I clicked on the generate image to prompt for an error
`Traceback (most recent call last):
  File "E:\Program Files\Fooocus_win64_1-1-10\Fooocus\modules\async_worker.py", line 902, in worker
    handler(task)
  File "E:\Program Files\Fooocus_win64_1-1-10\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\Program Files\Fooocus_win64_1-1-10\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\Program Files\Fooocus_win64_1-1-10\Fooocus\modules\async_worker.py", line 215, in handler
    cn_tasks[cn_type].append([cn_img, cn_stop, cn_weight])
KeyError: 0.6
Total time: 0.03 seconds`
I don't know what went wrong, could you please help me

**Comment by mashb1t** [comment]
this feature has been merged to develop and will be included in the next release. Feel free to directly check out develop so you don't accidentally miss something in the implementation.

**Pull request merged**